app/controllers/application_controller.rb

```diff
@@ -6,7 +6,9 @@ class ApplicationController < ActionController::Base
   protected
 
```

Comment:
- [ ] **[OPTIONAL]** I recommend adding this line of code to your application controller to ensure only logged-in users can use the application. There were a lot of unexpected issues because of the lack of this code. For example, when a user isn't logged in and they visit the post and comment `#new` pages they throw errors because there is no `current_user`

  ```
  before_action :authenticate_user!
  ```